File: 90-subsets-ii/subsets-ii.cpp
class Solution {
public:
    void generator(int i , vector<int>ans, set<vector<int>>&res, vector<int>&nums){
        if(i >= nums.size()){
            for (auto it : ans) cout << it << " ";
            cout << "\n";
            sort(ans.begin(),ans.end());
            res.insert(ans);
            return;
        }
        generator(i+1,ans,res,nums);
        ans.push_back(nums[i]);
        //ans.pop_back();
        generator(i+1,ans,res,nums);
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        set<vector<int>>res;
        vector<int>ans;
        generator(0,ans,res,nums);
        vector<vector<int>>resOne;
        for(auto i : res){
            resOne.push_back(i);
        }
        return resOne;
    }
};

File: 90-subsets-ii/subsets-ii.java
class Solution {
    public void creating(int[] nums, Set<List<Integer>>st, List<Integer>ls, int ind){
        if(ind == nums.length){
            st.add(new ArrayList<>(ls));
            return;
        }
        ls.add(nums[ind]);
        creating(nums, st, ls, ind + 1);
        ls.remove(ls.size() - 1);
        creating(nums, st, ls, ind + 1);
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        Set<List<Integer>>slist = new HashSet<>();
        List<Integer>lst = new ArrayList<>();
        creating(nums, slist, lst, 0);
        List<List<Integer>>res = new ArrayList<>();
        for (List<Integer> sl : slist) {
            res.add(new ArrayList<>(sl));
        }
        return res;
    }
}